src/views/studio/UploadEditVideoActionSheet/UploadEditVideoActionSheet.tsx

```diff
@@ -0,0 +1,436 @@
+            error={!!errors.title}
+            helperText={errors.title?.message}
+          />
+          <Textarea
+            name="description"
+            ref={register(textFieldValidation('Description', 0, 2160))}
+            maxLength={2160}
+            placeholder="Add video description"
+            error={!!errors.description}
+            helperText={errors.description?.message}
+          />
+          <FormField title="Video Visibility">
+            <Controller
+              name="selectedVideoVisibility"
+              control={control}
+              rules={requiredValidation('Video visibility')}
+              render={({ value }) => (
+                <Select
+                  value={visibilityOptions.find((o) => o.value === value) ?? null}
+                  items={visibilityOptions}
+                  onChange={(e) => {
+                    setFormValue('selectedVideoVisibility', e.selectedItem?.value)
+                    clearErrors('selectedVideoVisibility')
+                  }}
+                  error={!!errors.selectedVideoVisibility && !value}
+                  helperText={errors.selectedVideoVisibility?.message}
+                />
+              )}
+            />
+          </FormField>
+          <FormField title="Video Language">
+            <Controller
+              name="selectedVideoLanguage"
+              control={control}
+              rules={requiredValidation('Video language')}
+              render={({ value }) => (
+                <Select
+                  value={languages.find((l) => l.value === value) ?? null}
+                  items={[...languages]}
+                  onChange={(e) => {
+                    setFormValue('selectedVideoLanguage', e.selectedItem?.value)
+                    clearErrors('selectedVideoLanguage')
+                  }}
+                  error={!!errors.selectedVideoLanguage && !value}
+                  helperText={errors.selectedVideoLanguage?.message}
+                />
+              )}
+            />
+          </FormField>
+          <FormField title="Video Category">
+            <Controller
+              name="selectedVideoCategory"
+              control={control}
+              rules={requiredValidation('Video category')}
+              render={({ value }) => (
+                <Select
+                  value={
+                    (categories
+                      ?.map((category) => ({ name: category.name, value: category.id }))
+                      ?.find((c) => c.value === value) as SelectedItem) ?? null
+                  }
+                  items={
+                    (categories?.map((category) => ({ name: category.name, value: category.id })) as SelectedItem[]) ??
+                    []
+                  }
+                  onChange={(e) => {
+                    setFormValue('selectedVideoCategory', e.selectedItem?.value)
+                    clearErrors('selectedVideoCategory')
+                  }}
```

Comment:
I think you could move this to some handler, and make it more generic. This applies to almost all inputs.